### ISSUE TITLE
Arg_max Performance Regression

### ISSUE THREAD

**Issue by abstractqqq** [pola-rs/polars#11454]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

Try the following code on 0.18.3 and 0.19.5

```
df = pl.DataFrame({
    "test": np.random.randint(low = 0, high = 1000, size=1_000_000)
})
print(pl.show_versions())

%timeit df.select((pl.col("test") == pl.col("test").max()).arg_true().first().alias("1"))
%timeit df.select(pl.col("test").arg_max().alias("2"))
```

![old_polars](https://github.com/pola-rs/polars/assets/72534736/b31b3ccc-6ef6-4490-9828-0117cf622b87)
![new_polars](https://github.com/pola-rs/polars/assets/72534736/f2d34673-a9fb-45d4-96cd-c6ee692cbd92)

### Issue description

Arg_max performance has regressed by a lot since 0.19.5 ( mayber earlier )

### Expected behavior

Performance should stay the same

### Installed versions

<details>

```
--------Version info---------
Polars:              0.19.6
Index type:          UInt32
Platform:            Windows-10-10.0.19045-SP0
Python:              3.11.3 | packaged by Anaconda, Inc. | (main, Apr 19 2023, 23:46:34) [MSC v.1916 64 bit (AMD64)]


--------Version info---------
Polars:              0.18.3
Index type:          UInt32
Platform:            Windows-10-10.0.19045-SP0
Python:              3.11.3 | packaged by Anaconda, Inc. | (main, Apr 19 2023, 23:46:34) [MSC v.1916 64 bit (AMD64)]
```

</details>

**Comment by ritchie46** [comment]
I don't understand it. It hits exactly the same code branches and dispatches to the exact same fast path implemented here; https://github.com/jvdd/argminmax

We also use the same version. :thinking:

**Comment by abstractqqq** [comment]
I found that this only happens to integer columns (as in my 0.19.6 example above) and is mysteriously fixed in 0.19.7... I have no idea what is happening but I am glad it is fixed...

**Comment by ritchie46** [comment]
I believe this was a temporary regression in rustc. It got fixed, but we didn't change anything on the code, but we did update our compiler.
